### PR TITLE
[DEV-3202] add redirect fix and specs

### PR DESCRIPTION
**Pull request by ianhitchcock**
## Description of the change

Adds a fix to a phishing vulnerability in the managed portal's use of the `nextPage` url query parameter.

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation or Development tools (readme, specs, tests, code formatting)

## Links

- Jira issue number: [DEV-3202](https://saasquatch.atlassian.net/browse/DEV-3202)
- Process.st launch checklist: [HERE](https://app.process.st/runs/DEV-3202-Hosted-portal-phishing-vulnerability-fix-uZAHELG4j821wgf8c8ZAMw/tasks/ipJPSwLUytFe3xMt6P5PYw)

### Development

- [x] [Prettier](https://www.npmjs.com/package/prettier) was run (if applicable)
- [x] The behaviour changes in the pull request are covered by specs
- [ ] All tests related to the changed code pass in development

### Paperwork

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has a Jira number
- [x] This pull request has a Process.st launch checklist

### Code review

- [ ] Changes have been reviewed by at least one other engineer
- [ ] Security impacts of this change have been considered

### CODE REVIEW

**Review by loganvolkers — CHANGES_REQUESTED** [review]
Make more DRY

File: packages/mint-components/src/components/sqm-portal-reset-password/sqm-portal-reset-password.feature
Comment (loganvolkers) [inline packages/mint-components/src/components/sqm-portal-reset-password/sqm-portal-reset-password.feature:78]: I think that we should be allowing the search params to come through.

```suggestion
            | www.example.com | activity?foo=bar   | https://www.example.com/activity?foo=bar       |
            | www.example.com | /activity?foo=bar  | https://www.example.com/activity?foo=bar       |
```

File: packages/mint-components/src/utilities.ts
Comment (loganvolkers) [inline packages/mint-components/src/utilities.ts:40]: Allow `search` to be included as well instead of only path.

```suggestion
export function sanitizeUrlPath(path: string): URL {
  const url = new URL(path, window.location.href);
  const cleanUrl = new URL(window.location.href);
  cleanUrl.pathname = url.pathname;
  cleanUrl.search = url.search;
  return cleanUrl;
}
```

**Review by loganvolkers — CHANGES_REQUESTED** [review]
Should allow search, not just path

File: packages/mint-components/package.json
Comment (loganvolkers) [inline packages/mint-components/package.json:3]: Needs to be based on production

```suggestion
  "version": "1.5.1",
```

**Review by loganvolkers — CHANGES_REQUESTED** [review]
Looks good, but needs a changelog and the right version number.

Also I am sad that you didn't use jest-cucumber to automate the feature files.

File: packages/mint-components/package.json
Comment (DerekSiemens) [inline packages/mint-components/package.json:3]: Make sure to keep an eye on #265, if that goes out then this number will change again

Comment (ianhitchcock) [reply packages/mint-components/package.json]: I used 1.5.2 in anticipation of 1.5.1 being release first. I will wait for that release for a final rebase before asking for another PR approval

**Comment by ianhitchcock** [comment]
Passed QA: https://coda.io/d/Redirect-Security-Fix_dXCxBsJaF6y/QA_su8P-#_lu_TY

Changelog entry added and version number set to 1.5.2

File: packages/mint-components/CHANGELOG.md
Comment (DerekSiemens) [inline packages/mint-components/CHANGELOG.md:388]: This needs to be updated for the latest version 1.5.2...HEAD

File: packages/mint-components/CHANGELOG.md
Comment (DerekSiemens) [inline packages/mint-components/CHANGELOG.md:10]: Date is not correct if you are trying to deploy this today

File: packages/mint-components/CHANGELOG.md
Comment (ianhitchcock) [reply packages/mint-components/CHANGELOG.md]: 👍 fixed

File: packages/mint-components/CHANGELOG.md
Comment (ianhitchcock) [reply packages/mint-components/CHANGELOG.md]: 👍 fixed

**Review by loganvolkers — DISMISSED** [review]
Looks good to me

**Review by DerekSiemens — APPROVED** [review]

**Pull request merged**
